modules/sdk-coin-avaxp/src/lib/utxoEngine.ts

```diff
@@ -0,0 +1,117 @@
+import { DecodedUtxoObj, SECP256K1_Transfer_Output } from './iface';
+import { BN, Buffer as BufferAvax } from 'avalanche';
+import { Signature } from 'avalanche/dist/common';
+import utils from './utils';
+import { BuildTransactionError } from '@bitgo/sdk-core';
+import { StandardAmountInput, StandardTransferableInput } from 'avalanche/dist/common/input';
+
+export interface InputData {
+  amount: BN;
+  txidBuf: BufferAvax;
+  outputidx: BufferAvax;
+  signareIdx: number[];
+  signatures: Signature[];
+}
+
+/**
+ * Inputs can be controlled but outputs get reordered in transactions
+ * In order to make sure that the mapping is always correct we create an addressIndx which matches to the appropiate
+ * signatureIdx
+ * @param {TransferableInput[]} utxos as transaction ins.
+ * @returns the list of UTXOs
+ */
+export function recoverUtxos(utxos: StandardTransferableInput[]): DecodedUtxoObj[] {
+  return utxos.map((utxo) => {
+    const secpInput = utxo.getInput() as StandardAmountInput;
+
+    // use the same addressesIndex as existing ones in the inputs
+    const addressesIndex: number[] = secpInput.getSigIdxs().map((s) => s.toBuffer().readUInt32BE(0));
+
+    return {
+      outputID: SECP256K1_Transfer_Output,
+      outputidx: utils.cb58Encode(utxo.getOutputIdx()),
+      txid: utils.cb58Encode(utxo.getTxID()),
+      amount: secpInput.getAmount().toString(),
+      threshold: addressesIndex.length,
+      addresses: [], // this is empty since the inputs from deserialized transaction don't contain addresses
+      addressesIndex,
+    };
+  });
+}
+
+/**
+ * Convert Utxos into inputs data. Input Objects changes regarding chains. This method return a plain object to be mapped late in chain input.
+ * Sender is a list of owners address and utxo address must contains all of them.
+ * Signers is a list of sender cat it in threshold size. Firsts senders are the signers.
+ *
+ * Output always get reordered we want to make sure we can always add signatures in the correct location.
+ * Signatures array store signers address of the expected signature. Tx sign replace the address for the signature.
+ * So the location of the signatures is guaranteed.
+ *
+ * To find the correct location for the signature, we use the output's addresses to create the signatureIdx in the order that we desire
+ * 0: user key, 1: hsm key, 2: recovery key
+ *
+ * @param utxos
+ * @param sender array of addresses
+ * @param threshold number of signatures required
+ * @return {
+ *   inputs: InputData[];
+ *   amount: BN;
+ * } as total amount and inputs with signatures as signers address to be replaced.
+ */
+export function utxoToInput(
+  utxos: DecodedUtxoObj[],
+  sender: BufferAvax[],
+  threshold = 2
+): {
+  inputs: InputData[];
+  amount: BN;
+} {
+  // amount spent so far
+  let currentTotal: BN = new BN(0);
+
+  const inputs = utxos
+    .filter((utxo) => utxo && utxo.outputID === SECP256K1_Transfer_Output)
+    .map((utxo, i) => {
```

Comment:
`i` is not used

Reply:
done 
https://github.com/BitGo/BitGoJS/blob/6d023175be2d689b599bbc8e62e71c713fa7934a/modules/sdk-coin-avaxp/src/lib/utxoEngine.ts#L75